pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java

```diff
@@ -219,7 +219,7 @@ private String getSQLTypeQuery(boolean typoidParam) {
     // (keeping old behaviour of finding types, that should not be found without correct search
     // path)
     StringBuilder sql = new StringBuilder();
-    sql.append("SELECT typinput='array_in'::regproc as is_array, typtype, typname, pg_type.oid ");
+    sql.append("SELECT typinput='pg_catalog.array_in'::regproc as is_array, typtype, typname, pg_type.oid ");
```

Comment:
@davecramer , do you know if `pg_type.typarray != 0` answers that better?
`typarray` was added in 8.3: https://www.postgresql.org/docs/8.3/catalog-pg-type.html while we support 8.4+

Reply:
If typarray is not 0 then it identifies another row in pg_type, which is the "true" array type having this type as element. So it only says that there is an array type for this type.

Reply:
indeed.

Then PG 14 might bring new joy with its typsubscript and typelem (https://www.postgresql.org/docs/14/catalog-pg-type.html).
However, it looks like typelem has broader meaning than just an array.